terraform/variables.tf

```diff
@@ -0,0 +1,40 @@
+variable "profile" {
+  default = "default"
+}
+
+variable "region" {
+  default = "us-east-1"
+}
+
+variable "aws_ecs_ami" {
+  # ECS-optimized AMI for us-east-1.
+  default = "ami-b2df2ca4"
+}
+
+variable "ecs_instance_profile" {
+  default = "arn:aws:iam::896538046175:instance-profile/terraform-wzxkyowirnachcosiqxrriheki"
+}
+
+variable "ecs_service_role" {
+  default = "arn:aws:iam::896538046175:role/ecs_service_role"
+}
+
+variable "subnet_id" {
```

Comment:
The software defined network that is used to house the compute resources within an AWS account is referred to as a Virtual Private Cloud (VPC). VPCs have a private IP address space that is carved up into subnets, route tables, and gateways.

![image](https://cloud.githubusercontent.com/assets/43639/22959473/a8a30b28-f304-11e6-8cc4-cc1ded0434db.png)

This diagram from the VPC docs shows a single VPC that is carved up into two subnets (one public and one private). The public subnet contains EC2 instances with public IPs and a NAT device. The private subnet also contains EC2 instances, but those only have private IPs, and when they need to hit the internet, they need to be routed through the NAT in the public subnet.

In our case, we want:

- EC2 instances that are members of the ECS cluster to live within a private subnet
- ALB instances (transparent to us) in the public subnet

The `aws_alb` attribute `subnets` is how we associate an ALB with specific subnets. `vpc_zone_identifier` of `aws_autoscaling_group` is how we associate the same for ASGs.

I took a look at the default VPC in the GeoTrellis AWS account and it has a bunch of subnets, but the route tables are not setup to handle routing for private subnets that need to interact with the internet (no route table entries, no NAT device).

We usually lay down a new VPC for each deployment:

- https://github.com/azavea/raster-foundry-deployment/blob/develop/terraform/network.tf#L4-L19
- https://github.com/azavea/terraform-aws-vpc

In this case, we can opt to do that, or manually tweak the default VPC for the GeoTrellis AWS account so that it can support both public and private subnets.